crates/agglayer-certificate-orchestrator/src/error.rs

```diff
@@ -34,8 +34,18 @@ pub enum CertificationError {
     Deserialize { source: bincode::Error },
     #[error("internal error: {0}")]
     InternalError(String),
+    #[error("prover failed")]
+    ProverFailed(String),
+    #[error("prover returned unspecified error")]
+    ProverReturnedUnspecifiedError,
+    #[error("prover execution failed: {source}")]
+    ProverExecutionFailed { source: ProofError },
```

Comment:
`source` should not be in the error message, as long as we don't rely on it for backwards compat somewhere (which I have not checked)
```suggestion
    #[error("prover execution failed")]
    ProverExecutionFailed { source: ProofError },
```